### PR TITLE
Update deviceController to reconcile the device between the openyurt and edge

### PR DESCRIPTION
**Pull request by qclc**
1. Update deviceController to reconcile the device between the OpenYurt and EdgeX Fountry, including:
    - Synchronize device objects on edge platform to the OpenYurt cluster
    - Openyurt creates and deletes device objects that can be synchronized to the edge platform
    - To support declarative device state modification, i.e., modifying the device's properties by changing fields of the device CRs 
    (Note: only support the `DeviceProperties`, `AdminState` and `OperatingState` fields of device)
2. Change the reference repository from `charleszheng44` to `openyurtio`

### CODE REVIEW

File: clients/edgex-foundry/device_client.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
+package edgex_foudry
```
Comment (rambohe-ch) [inline clients/edgex-foundry/device_client.go:1]: please add license header

File: clients/edgex-foundry/device_client.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+}
+
+type EdgexDeviceClient struct {
+	*resty.Client
+	CoreMetaClient    ClientURL
+	CoreCommandClient ClientURL
+	logr.Logger
+}
+
+func NewEdgexDeviceClient(coreMetaClient, coreCommandClient ClientURL, log logr.Logger) *EdgexDeviceClient {
```
Comment (rambohe-ch) [inline clients/edgex-foundry/device_client.go:30]: why use `logr.Logger` instead of `k8s.io/klog`?

File: clients/edgex-foundry/util.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
+package edgex_foudry
```
Comment (rambohe-ch) [inline clients/edgex-foundry/util.go:1]: please add license header

File: clients/errors.go [hunk 4]
```diff
@@ -8,6 +8,8 @@ type NotFoundError struct{}
 func (e *NotFoundError) Error() string { return "Item not found" }
 
 func IsNotFoundErr(err error) bool {
-	return err.Error() == "Item not found" || strings.HasPrefix(err.Error(), "no item found")
-	// return errors.Is(err, &NotFoundError{})
+	if err == nil {
```
Comment (rambohe-ch) [inline clients/errors.go:11]: please add license header

File: clients/interface.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 The Kubernetes authors.
```
Comment (rambohe-ch) [inline clients/interface.go:2]: Kubernetes author --> OpenYurt author?

File: clients/edgex-foundry/device_client.go (hunk 1 above)
Comment (qclc) [reply clients/edgex-foundry/device_client.go:17]: Thanks for reminding, the license header is added.

File: clients/edgex-foundry/util.go (hunk 3 above)
Comment (qclc) [reply clients/edgex-foundry/util.go:17]: Thanks for reminding, the license header is added.

File: clients/errors.go (hunk 4 above)
Comment (qclc) [reply clients/errors.go:27]: Thanks for reminding, the license header is added.

File: clients/interface.go (hunk 5 above)
Comment (qclc) [reply clients/interface.go]: Thanks for reminding, the license header is changed.

File: clients/edgex-foundry/device_client.go (hunk 2 above)
Comment (qclc) [reply clients/edgex-foundry/device_client.go:41]: Thanks for your suggestion, and I will submit a PR version to replace all `logr.Logger` with `k8s.io/klog`

**Comment by rambohe-ch** [comment]
@qclc please fix the conflicting files.

**Comment by qclc** [comment]
@rambohe-ch all the conflicting files are fixed, please review.

**Review by rambohe-ch — APPROVED** [review]
LGTM

**Pull request merged**